xgi/classes/reportviews.py

```diff
@@ -38,7 +38,14 @@ class IDView(Mapping, Set):
 
     """
 
-    __slots__ = ("_dispatcher", "_id_dict", "_id_attr", "_ids")
+    __slots__ = (
+        "_dispatcher",
+        "_id_dict",
+        "_id_attr",
+        "_bi_id_dict",
+        "_bi_id_attr",
+        "_ids",
```

Comment:
At this point we might want to consider (in the future) just passing the entire network...

Reply:
haha, good point. In the future perhaps the whole network and an indicator whether it's for nodes or edges?